symmer/symplectic/base.py

```diff
@@ -1249,3 +1235,122 @@ def from_array(cls,
         coeff_vector = statevector[non_zero]
         Qstate = cls(state_matrix, coeff_vector, vec_type=vec_type)
         return Qstate
+
+    def plot_state(self, logscale:bool = False):
+
+        assert self._is_normalized(), 'should only plot normalized quantum states'
+
+        # clean duplicate states and set amplitdue threshold
+        q_state = self.cleanup()
+        prob = np.abs(q_state.state_op.coeff_vec) ** 2
+
+        fig, ax = plt.subplots(1, 1)
+
+        if q_state.state_op.n_qubits<64:
+            x_binary_ints = q_state.state_matrix @ (1 << np.arange(q_state.state_matrix.shape[1])[::-1])
+        else:
+            x_binary_ints = q_state.state_matrix @ (1 << np.arange(q_state.state_matrix.shape[1], dtype=object)[::-1])
+
+        if prob.shape[0]<2**8:
+            # bar chart
+            ax.bar(x_binary_ints, prob, width=1, edgecolor="white", linewidth=0.8)
+        else:
+            # line plot
+            sort_inds = np.argsort(x_binary_ints)
+            x_data = x_binary_ints[sort_inds]
+            y_data = prob[sort_inds]
+            ax.plot(x_data, y_data)
+        ax.set(xlabel='binary output', ylabel='probability amplitude')
+
+        if logscale:
+            ax.set_yscale('log')
+
+        return (ax)
+
+
+
+def random_anitcomm_2n_1_PauliwordOp(n_qubits, complex_coeff=True, apply_clifford=True):
+    """ Generate a anticommuting PauliOperator of size 2n+1 on n qubits (max possible size)
+        with normally distributed coefficients. Generates in structured way then uses Clifford rotation (default)
+        to try and make more random (can stop this to allow FAST build, but inherenet structure
+         will be present as operator is formed in specific way!)
+    """
+    base = 'X' * n_qubits
+    I_term = 'I' * n_qubits
+
+    P_list = [base]
+    for i in range(n_qubits):
+        # Z_term
+        P_list.append(base[:i] + 'Z' + I_term[i + 1:])
+        # Y_term
+        P_list.append(base[:i] + 'Y' + I_term[i + 1:])
+
+    coeff_vec = np.random.randn(len(P_list)).astype(complex)
+    if complex_coeff:
+        coeff_vec += 1j * np.random.randn((len(P_list)))
+
+    P_anticomm = PauliwordOp.from_dictionary((dict(zip(P_list, coeff_vec))))
+
+    # random rotations to get rid of structure
+    if apply_clifford:
+        for _ in range(10):
+            P_rand = PauliwordOp.random(n_qubits, 1, complex_coeffs=complex_coeff)
+            P_rand.coeff_vec[0] = 1
+            P_anticomm = P_anticomm._rotate_by_single_Pword(P_rand,
+                                                            None)
+
+    anti_comm_check = P_anticomm.adjacency_matrix.astype(int) - np.eye(P_anticomm.adjacency_matrix.shape[0])
+    assert(np.sum(anti_comm_check) == 0), 'operator needs to be made of anti-commuting Pauli operators'
+
+    return P_anticomm
+
+
+def get_computational_projector(N_qubits:int,
```

Comment:
Build a PauliwordOp projecting certain qubit indices onto fixed qubit states!

Reply:
Nice! To be moved to symplectic.utils